### DISCUSSION TITLE
23827 수열 (Easy) 문제 접근에 대한 키워드 혹은 유사한 문제

### DISCUSSION THREAD

**Discussion by hyeonic** [woowacourse-study/2022-lv3-algorithm-study#44] [알고리즘 토론할래요]
23827 문제에서 최대 나올 수 있는 수의 개수가 50만개로 케이가 언급한 것 처럼 이중 for문으로 해결할 경우 시간초과가 날 것으로 추측 되는데요!
홀로 고민한 이후 구글링을 통해 힌트를 얻었습니다. 만약 `a, b, c, d, e`와 같은 수열이 주어졌다고 가정하고 일반화를 진행했을 때 아래와 같이 정리 된다고 합니다.  

```
a x (b + c + d + e) + b x (c + d + e) + c x (d + e) + d x (e) 
```

문제만 보고 위와 같은 형태를 떠오르기 힘들었기 때문에 많은 고민을 동반하게 되었습니다. 문제 접근을 위해서 관련된 키워드나 알고리즘 카테고리가 있을까요? 혹은 비슷한 유형의 문제가 무엇이 있을까요? 해당 문제만 보고 어떠한 방식을 유도한 것인지 알아 차라는 것이 쉽지 않네요 ㅜㅠ

**Comment by nbalance97** [comment]
관련 키워드로는 수학인데.. https://solved.ac/problems/tags/math
개인적인 생각으로는 수학 파트의 경우는 구글링이 필요한 부분도 있다고 생각이 듭니다..

경험담을 말씀드리면 소수를 구하는 방법 중 에라토스테네스의 체를 사용해야 하는 문제가 있었습니다. 처음에는 이런거를 어떻게 알아요..ㅋㅋㅋ
구글링을 해서 풀이 방식만 보고, 이걸 코드로 옮기는 과정을 해보는 것 자체로도 어느정도 도움이 될 것 같아요.

**Comment by kth990303** [comment]
누적합을 캐싱해서 빠르게 푸는 쪽도 어려운데, 이 부분은 누적합 알고리즘을 공부해보면 좋을 것 같아요.
